src/mlpack/methods/ann/layer/binary_rbm_impl.hpp

```diff
@@ -19,61 +19,63 @@ namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
 template<typename InputDataType, typename OutputDataType>
-VisibleLayer<InputDataType, OutputDataType>::VisibleLayer(
+BinaryLayer<InputDataType, OutputDataType>::BinaryLayer(
   const size_t inSize,
-	const size_t outSize):
+	const size_t outSize,
+  const bool typeVisible):
   inSize(inSize), 
-  outSize(outSize)
+  outSize(outSize),
+  typeVisible(typeVisible)
 {
   weights.set_size(outSize * inSize + inSize, 1);
 }
 
 template<typename InputDataType, typename OutputDataType>
-void VisibleLayer<InputDataType, OutputDataType>::Reset()
+void BinaryLayer<InputDataType, OutputDataType>::Reset()
 {
-  
-  weight = arma::mat(weights.memptr(), outSize, inSize, false, false);
-  ownBias = arma::mat(weights.memptr() + weight.n_elem, inSize, false, false);
-  otherBias = arma::mat(weight.memptr() + weight.n_elem + inSize, outSize, false, false);
-
+  if(typeVisible)
```

Comment:
Do you mean to say here that when we are initializing the 1.hidden layer(outsize, insize, 0).
According to that i think you might be correct but i wrote it with the idea that 2.hidden layer(insize, outsize, 0). Should i change this to keeping the 1 in mind.